vars/uploadMinSnapshotsToS3.groovy

```diff
@@ -1,15 +1,42 @@
 void call(Map args = [:]) {
+
     def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    List<Closure> fileActions = args.fileActions ?: []
     String manifest = args.manifest ?: "manifests/${INPUT_MANIFEST}"
+
     def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifest))
     echo("Retreving build manifest from: $WORKSPACE/builds/${inputManifest.build.getFilename()}/manifest.yml")
-    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: "$WORKSPACE/builds/${inputManifest.build.getFilename()}/manifest.yml"))
+
+    productName = inputManifest.build.getFilename()
+    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: "$WORKSPACE/builds/${productName}/manifest.yml"))
     version = buildManifest.build.version
     architecture = buildManifest.build.architecture
     platform = buildManifest.build.platform
-    minArtifactSourceName = "opensearch-min-${version}-${platform}-${architecture}.tar.gz"
-    minArtifactName = "opensearch-min-${version}-${platform}-${architecture}-latest.tar.gz"
+    id = buildManifest.build.id
+
+    // Setup src & dst variables for artifacts
+    srcDir = "${WORKSPACE}/builds/${productName}/dist"
+    dstDir = "snapshots/core/${productName}/${version}"
+    baseName = "${productName}-min-${version}-${platform}-${architecture}"
+    tarSrcName = "${baseName}.tar.gz"
+    tarDstNameId = "${baseName}-${id}.tar.gz"
```

Comment:
Nit: I find additional variables to be overhead that makes code harder to follow. The ones like `tarDstNameId` are only used in 1 place below, I would directly use the value there. No need to change anything unless you agree, just something to think about.

Reply:
Will try to make these simpler, tbh I also get a bit confused myself on this.

Reply:
fixed.